codalab/lib/beam/MultiReaderFileStream.py

```diff
@@ -1,21 +1,25 @@
-from io import BytesIO
+from io import BytesIO, SEEK_SET, SEEK_END
 from threading import Lock
-
-from codalab.worker.un_gzip_stream import BytesBuffer
-
-
+import time
 class MultiReaderFileStream(BytesIO):
     """
-    FileStream that support multiple readers
+    FileStream that supports N readers with the following features and constraints:
```

Comment:
Should say something even more general like: Given an input stream `fileobj`, copy this to multiple streams (readers) that have to be put in sync.